nntrainer/tensor/hgemm/hgemm_kernel/hgemm_kernel_8x16_experimental.cpp

```diff
@@ -0,0 +1,799 @@
+// SPDX-License-Identifier: Apache-2.0
+/**
+ * Copyright (C) 2024 Sungsik Kong <ss.kong@samsung.com>
+ *
+ * @file   hgemm_kernel_8x16.cpp
+ * @date   04 April 2024
+ * @see    https://github.com/nnstreamer/nntrainer
+ * @author Sungsik Kong <ss.kong@samsung.com>
+ * @bug    No known bugs except for NYI items
+ * @brief  This is half-precision GEMM 8x16 kernel
+ *
+ */
+
+#include <arm_neon.h>
+#include <assert.h>
+#include <hgemm_kernel.h>
+#include <stdlib.h>
+
+#define INIT_KERNEL_8X16()       \
+  do {                           \
+    v0_7 = vdupq_n_f16(0.F);     \
+    v8_15 = vdupq_n_f16(0.F);    \
+    v16_23 = vdupq_n_f16(0.F);   \
+    v24_31 = vdupq_n_f16(0.F);   \
+    v32_39 = vdupq_n_f16(0.F);   \
+    v40_47 = vdupq_n_f16(0.F);   \
+    v48_55 = vdupq_n_f16(0.F);   \
+    v56_63 = vdupq_n_f16(0.F);   \
+    v64_71 = vdupq_n_f16(0.F);   \
+    v72_79 = vdupq_n_f16(0.F);   \
+    v80_87 = vdupq_n_f16(0.F);   \
+    v88_95 = vdupq_n_f16(0.F);   \
+    v96_103 = vdupq_n_f16(0.F);  \
+    v104_111 = vdupq_n_f16(0.F); \
+    v112_119 = vdupq_n_f16(0.F); \
+    v120_127 = vdupq_n_f16(0.F); \
+  } while (0)
+
+// 1. Partial sum 2048 digits
+#define KERNEL_8x16_ACC16()                            \
```

Comment:
I think this entire macro can be reduced.

```suggestion
for (int i = 0; i < 15; ++i) {
  va0 = vld1q_f16(a + 8 * n);
  vb1 = vld1q_f16(b + 8 * n * 2);
  vb2 = vld1q_f16(b + 8 * (n * 2 + 1));
  v0_7 = vfmaq_laneq_f16(v0_7, vb1, va0, 0);
  v8_15 = vfmaq_laneq_f16(v8_15, vb1, va0, 1);
  v16_23 = vfmaq_laneq_f16(v16_23, vb1, va0, 2);
  v24_31 = vfmaq_laneq_f16(v24_31, vb1, va0, 3);
  v32_39 = vfmaq_laneq_f16(v32_39, vb1, va0, 4);
  v40_47 = vfmaq_laneq_f16(v40_47, vb1, va0, 5);
  v48_55 = vfmaq_laneq_f16(v48_55, vb1, va0, 6);
  v56_63 = vfmaq_laneq_f16(v56_63, vb1, va0, 7);
  v64_71 = vfmaq_laneq_f16(v64_71, vb2, va0, 0);
  v72_79 = vfmaq_laneq_f16(v72_79, vb2, va0, 1);
  v80_87 = vfmaq_laneq_f16(v80_87, vb2, va0, 2);
  v88_95 = vfmaq_laneq_f16(v88_95, vb2, va0, 3);
  v96_103 = vfmaq_laneq_f16(v96_103, vb2, va0, 4);
  v104_111 = vfmaq_laneq_f16(v104_111, vb2, va0, 5);
  v112_119 = vfmaq_laneq_f16(v112_119, vb2, va0, 6);
  v120_127 = vfmaq_laneq_f16(v120_127, vb2, va0, 7);
}
```